### PR TITLE
Add water heater data for IECC

### PR DESCRIPTION
**Pull request by lzwang26**
Added water heater data for IECC.

### CODE REVIEW

File: database_files/hvac_minimum_requirement_water_heaters_90_1.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 [
     {
         "template": "90.1-2004",
```
Comment (lymereJ) [inline database_files/hvac_minimum_requirement_water_heaters_90_1.json]: @lzwang26  - Could you please revert the changes in this file? They are going to create conflicts when #2 is merged.

Comment (lzwang26) [reply database_files/hvac_minimum_requirement_water_heaters_90_1.json]: Finished the reverting. Thank you!

**Review by lymereJ — APPROVED** [review]
